### PR TITLE
Updating dependencies to fix the CI

### PR DESCRIPTION
**Pull request by devinmatte**
# The issue

CI is failing now due to a broken peer dependency with `reakit` requiring `react` 16 or 17 (we use 18 now).

# Solutions

- Revert back to React 17
- Migrate to the alpha version of `ariakit`
    - `reakit` has been renamed in v2

# Changes
- Updates a few dependencies related to `eslint` and `webpack`
- Moves types to `devDependencies` where they belong (ts types aren't included in resulting bundle)

# Motivation

- Want to sync eslint configs between repos in a future PR, in order to do that, I want to get eslint on this repo up to date
- Fix CI on other PRs/commits

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,39 +30,39 @@
...
   "dependencies": {
-    "@types/react": "^18.0.1",
-    "@types/react-dom": "^18.0.0",
     "bezier-js": "^2.5.1",
     "classnames": "^2.2.6",
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0",
-    "react-favicon": "1.0.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
```
Comment (thiteixeira) [inline package.json:36]: Can we just use React 17 for now? I don't think we are using any features from React 18

Comment (devinmatte) [reply package.json:36]: Yeah. That's the easier solution lol. I should just do that

**Review by thiteixeira — APPROVED** [review]
Looks good. 
Git is complaining about a missing dependency on a useEffect. Should we fix it here?

Screen Shot 2022-08-07 at 2 29 08 PM

**Comment by devinmatte** [comment]
I think that may be intentionally missing, I don't think we'd want to listed for the function updates. Since it's a warning we probably don't need to do anything about it now.

**Comment by thiteixeira** [comment]
Probably just needs a `// eslint-disable-next-line react-hooks/exhaustive-deps` as well then. But optional for this PR

**Pull request merged**